File: app/Repositories/Gtfs/Models/PickupDropoffType.php
<?php

namespace Irail\Repositories\Gtfs\Models;

enum PickupDropoffType: int
{
    /**
     * Regularly scheduled pickup/dropoff.
     */
    case    ALWAYS = 0;

    /**
     * No pickup/dropoff available.
     */
    case    NEVER = 1;

    /**
     * Must phone agency to arrange pickup/dropoff.
     */
    case    BOOKING_REQUIRED = 2;

    /**
     * Must coordinate with driver to arrange pickup/dropoff.
     */
    case    REQUEST_STOP = 3;

}

File: src/main/java/be/irail/api/gtfs/reader/models/PickupDropoffType.java
package be.irail.api.gtfs.reader.models;

public enum PickupDropoffType {
    SCHEDULED(0),
    NONE(1),
    PHONE_TO_ARRANGE(2),
    COORDINATE_WITH_DRIVER(3);


    private final int code;

    PickupDropoffType(int code) {
        this.code = code;
    }

    public static PickupDropoffType fromCode(int code) {
        for (PickupDropoffType type : values()) {
            if (type.code == code) {
                return type;
            }
        }
        return null;
    }
}
